src/Symfony/Component/Form/Extension/Core/DataMapper/CheckboxListMapper.php

```diff
@@ -46,22 +35,12 @@ public function mapDataToForms($choices, $checkboxes)
         }
 
         if (!is_array($choices)) {
-            throw new TransformationFailedException('Expected an array.');
-        }
-
-        try {
-            $valueMap = array_flip($this->choiceList->getValuesForChoices($choices));
-        } catch (\Exception $e) {
-            throw new TransformationFailedException(
-                'Can not read the choices from the choice list.',
-                $e->getCode(),
-                $e
-            );
+            throw new UnexpectedTypeException($choices, 'array');
         }
 
         foreach ($checkboxes as $checkbox) {
             $value = $checkbox->getConfig()->getOption('value');
-            $checkbox->setData(isset($valueMap[$value]) ? true : false);
+            $checkbox->setData(in_array($value, $choices, true) ? true : false);
```

Comment:
I would indeed remove those unneeded ternary operators when the left side is already a bool.